src/controller/userController.js

```diff
@@ -288,3 +288,66 @@ export const userSignin = (req, res, next) => {
       });
     });
 };
+
+/**
+ *
+ * @api {post} /api/v1/users/:userId Login an existing user
+ * @apiName ViewUserProfile
+ * @apiGroup User
+ *
+ * @apiSuccessExample Success Response
+ * HTTP/1.1 200 OK
+ * {
+ *   "status": "success",
+ *  "data": {
+ *     "message": "Authentication successful",
+ *     "user": {
+ *        "firstName":"John",
+ *        "lastName":"Doe",
+ *        "email":"johndoe@mymail.com",
+ *        "password":"johndoe",
+ *        "phoneNumber":"098654321",
+ *        "userType":"Producer",
+ *        "businessName":"My Business",
+ *        "bio":"My business biography",
+ *        "address":"No 2 Lokoja Stree, Abuja. FCT",
+ *        "photo": 'http://link-to-file'
+ *      }
+ *   }
+ * }
+ *
+ * @apiError Authentication Failed
+ *
+ * @apiErrorExample Error Response:
+ * HTTP/1.1 401 Unauthorized
+ * {
+ *    "status": "error",
+ *    "data": {
+ *      "message":"Authentication Failed",
+ *      error: "Error information"
+ *    }
+ * }
+ */
+
+export const getUserProfile = async (req, res, next) => {
+  const { userId } = req.params;
+  try {
+    // Find the user by primary key.
+    const user = await User.findByPk(userId);
```

Comment:
A none existing userId in the database still returns a 200 O.K "User found" with user data null.  An "If" check could be performed and return error "user not found if userId doesn't exist.

Reply:
Thank you, have updated this.